drivers/wdf/wdfldr/common.c

```diff
@@ -0,0 +1,826 @@
+		}
+		totalSize = 0;
+
+		for(;;)
+		{
+			numberOfBytes = 0;
+			status = AuxKlibQueryModuleInformation(&numberOfBytes, sizeof(RTL_MODULE_EXTENDED_INFO), 0);
+
+			if (!NT_SUCCESS(status) || !numberOfBytes)
+			{
+				if (!WdfLdrDiags)
+					goto clean;
+				break;
+			}
+			status = RtlULongAdd(numberOfBytes, totalSize, &informationLength);
+
+			if (!NT_SUCCESS(status)) 
+			{
+				if (WdfLdrDiags) 
+				{
+					DbgPrint("WdfLdr: GetImageBase - ");
+					DbgPrint("ERROR: RtlUlongAdd failed with Status 0x%x\n", status);
+				}
+			}
+			else 
+			{
+				numberOfBytes = informationLength;
+			}
+
+			infoBuffer = ExAllocatePoolWithTag(PagedPool, numberOfBytes, WDFLDR_TAG);
+			pModuleInfo = infoBuffer;
+
+			if (pModuleInfo == NULL)
+			{
+				status = STATUS_INSUFFICIENT_RESOURCES;
+
+				if (WdfLdrDiags) 
+				{
+					DbgPrint("WdfLdr: GetImageBase - ");
+					DbgPrint("ERROR: ExAllocatePoolWithTag failed with Status 0x%x\n", status);
+				}
+				goto end;
+			}
+
+			memset(pModuleInfo, 0, numberOfBytes);
+			informationLength = numberOfBytes;
+			status = AuxKlibQueryModuleInformation(&informationLength, sizeof(RTL_MODULE_EXTENDED_INFO), pModuleInfo);
+
+			if (status != STATUS_BUFFER_TOO_SMALL)
+				break;
+
+			ExFreePoolWithTag(pModuleInfo, 0);
+			totalSize += sizeof(RTL_MODULE_EXTENDED_INFO);
+			pModuleInfo = NULL;
+
+			if (totalSize >= sizeof(RTL_MODULE_EXTENDED_INFO) * 10)
+				goto clean;
+		}
+
+		if (!NT_SUCCESS(status))
+		{
+			if (!WdfLdrDiags)
+				goto end;
+
+			DbgPrint("WdfLdr: GetImageBase - ");
+			DbgPrint("ERROR: AuxKlibQueryModuleInformation failed with Status 0x%x\n", status);
+			goto end;
+		}
+
+		endOfArray = (ULONG_PTR)pModuleInfo + informationLength;
+		numberOfBytes = informationLength;
+
+		for(;;)
+		{
+			if (pModuleInfo->FileNameOffset < 0x100 &&
+				strncmp(pModuleInfo->FullPathName, fileName, fileNameLength) == 0)
+				break;
+
+			pModuleInfo++;
+			if ((ULONG_PTR)pModuleInfo >= endOfArray)
+				goto end;
+		}
+		*ImageBase = pModuleInfo->BasicInfo.ImageBase;
+		*ImageSize = pModuleInfo->ImageSize;
+	}
+	else 
+	{
+		if (WdfLdrDiags) 
+		{
+			DbgPrint("WdfLdr: GetImageBase - ");
+			DbgPrint("ERROR: RtlUnicodeStringToAnsiString failed with Status 0x%x\n", status);
+		}
+		ansiImageName.Buffer = NULL;
+	}
+end:
+	if (infoBuffer != NULL)
+		ExFreePoolWithTag(infoBuffer, 0);
+clean:
+	if (ansiImageName.Buffer)
+		RtlFreeAnsiString(&ansiImageName);
+
+	return status;
+}
+
+
+BOOLEAN
+NTAPI
+ServiceCheckBootStart(
+	IN PUNICODE_STRING Service
+)
+{
+	NTSTATUS status;
+	OBJECT_ATTRIBUTES ObjectAttributes;
+	HANDLE KeyHandle;
+	BOOLEAN result;
+	ULONG value;
+	UNICODE_STRING ValueName;
+
+	KeyHandle = NULL;
+	result = FALSE;
+	ObjectAttributes.Length = 24;
+	ObjectAttributes.RootDirectory = 0;
+	ObjectAttributes.Attributes = OBJ_KERNEL_HANDLE | OBJ_CASE_INSENSITIVE;//576;
+	ObjectAttributes.ObjectName = Service;
+	ObjectAttributes.SecurityDescriptor = 0;
+	ObjectAttributes.SecurityQualityOfService = 0;
```

Comment:
Can you use the appropriate macro InitializeObjectAttributes() and avoid magic value?